merkledag.go

```diff
@@ -2,50 +2,29 @@ package format
 
 import (
 	"context"
-	"errors"
 
 	cid "github.com/ipfs/go-cid"
+	"github.com/ipld/go-ipld-prime/storage"
 )
 
 // ErrNotFound is used to signal when a Node could not be found. The specific
 // meaning will depend on the DAGService implementation, which may be trying
 // to read nodes locally but also, trying to find them remotely.
-//
-// The Cid field can be filled in to provide additional context.
-type ErrNotFound struct {
-	Cid cid.Cid
-}
-
-// Error implements the error interface and returns a human-readable
-// message for this error.
-func (e ErrNotFound) Error() string {
-	if e.Cid == cid.Undef {
-		return "ipld: could not find node"
-	}
-
-	return "ipld: could not find " + e.Cid.String()
-}
+type ErrNotFound = storage.ErrNotFound
```

Comment:
> I'd like to make progress with this if possible but don't want to upset people by forging ahead if you disagree with the approach.

Thanks @rvagg I appreciate you both pushing on this and the thought here.

---

Some concerns/comments here:

### NotFound() bool being too generic

There are a bunch of NotFound errors in the Go ecosystem that would match the check in the go-ipld-prime PR https://grep.app/search?q=%20notfound%28%29%20bool&filter[lang][0]=Go. Depending on how these errors are being used this could lead to bugs.

### Adding the go-ipld-prime dependency

AFAICT the only change in this PR is to move where the error type lives since it could just as easily live on go-ipld-format as go-ipld-prime. The go-ipld-prime PR adds the changes which are mostly modifying `Is()` to apply to any `NotFound() bool`. If so does that just mean that we're at "go-ipld-prime doesn't want to import go-ipld-format, and go-ipld-format doesn't want to import go-ipld-prime"? If so it doesn't appear that there's much real benefit in moving this error.

> In the meantime, I think this PR at least puts things on equal footing. It is super frustrating to HAVE to include this library to get a not found error to be recognized by our stack. Especially if you are using go-ipld-prime.

This ends up being false, since the PR just inverts which package would need to depend on which.

I don't personally care a ton here, since it's not like importing a couple variables from a package and re-exporting them is going to be a serious problem if the package is small and doesn't import too many dependencies, but it we're at the point of debating which library the error should live in I'm calling out that neither is automatically better.

---

Maybe it does end up being worthwhile (as was mentioned earlier in this PR) to create a new package. If it offends people's sensibilities to put the new package in either of go-ipld-format or go-ipld-prime then another repo is doable too. IMO this isn't going to be the last time we're going to be dealing with these kinds of error inconsistencies.

For example, in some recent gateway work in boxo we've ended up with https://github.com/ipfs/boxo/blob/1ec05d754868ab80af629b4a9401ee575cecbe1e/gateway/errors.go#L180-L208 which is pretty annoying and something that may pop up for other IPLD users.

In this case we'd end up with at least `NotFound` and `InvalidTraversal` errors and we'd make them IPLD specific.

Thoughts?